abci/client/client.go

```diff
@@ -115,18 +116,25 @@ func (r *ReqRes) InvokeCallback() {
 	defer r.mtx.Unlock()
 
 	if r.cb != nil && r.Response != nil {
-		r.cb(r.Response)
+		r.cbErr = r.cb(r.Response)
 	}
 	r.callbackInvoked = true
 }
 
+// Error returns the error returned by the callback, if any.
+func (r *ReqRes) Error() error {
+	return r.cbErr
+}
+
 // GetCallback returns the configured callback of the ReqRes object which may be
 // nil. Note, it is not safe to concurrently call this in cases where it is
 // marked done and SetCallback is called before calling GetCallback as that
 // will invoke the callback twice and create a potential race condition.
 //
 // ref: https://github.com/tendermint/tendermint/issues/5439
-func (r *ReqRes) GetCallback() func(*types.Response) {
+//
+// Deprecated: not used.
```

Comment:
Why not remove it?

Reply:
Wouldn't that be API breaking? If yes, I'd remove it in the next major release.

Reply:
It is, but it's okay to break API in `v1` and `main`

Reply:
Ok, I removed it.

Reply:
Actually, since this is the only breaking part in this PR, I restored the method and created another PR to remove it #4083.